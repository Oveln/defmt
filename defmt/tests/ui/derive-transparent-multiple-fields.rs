#[derive(defmt::Format)]
#[defmt(transparent)]
enum A {
    Invalid { foo: u8, bar: i8 },
}

#[derive(defmt::Format)]
#[defmt(transparent)]
struct Foo {
    foo: u8,
    bar: i8,
}

fn main() {}
