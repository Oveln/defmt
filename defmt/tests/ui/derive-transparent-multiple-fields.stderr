error: Transparent format can only be applied when all variants have exactly one field.
 --> tests/ui/derive-transparent-multiple-fields.rs:1:10
  |
1 | #[derive(defmt::Format)]
  |          ^^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `defmt::Format` (in Nightly builds, run with -Z macro-backtrace for more info)
